pkg/validation/application_installation.go

```diff
@@ -69,6 +69,13 @@ func ValidateApplicationInstallationSpec(ctx context.Context, client ctrlruntime
 		}
 	}
 	allErrs = append(allErrs, ValidateDeployOpts(spec.DeployOptions, specPath.Child("deployOptions"))...)
+
+	// Ensure that not both values and ValuesBlock fields are set simultaneously
+	if len(ai.Spec.Values.Raw) > 0 && ai.Spec.ValuesBlock != "" {
```

Comment:
The check makes sense but what will the workflow look like for an AppIns that had `values` configured but is now being modified from the UI? The new value with be persisted in `values` or `valuesBlock`. If it's in `values` then we need to inform users that they need to migrate to `valuesBlock` to preserve comments.

Reply:
My proposal for this would be to do a conversion in the FE. By this I mean:
The UI reads in values and valuesBlock and loads the content of the field of the two that is filled in the yaml editor. Whenever the UI is sending a POST/PUT for an AppInstall, it sends an empty values field and a valuesBlock field with the content of the yaml editor. Thus doing a "conversion" for AppInstalls with the old field filled. I think this should be safe: because GitOps users will keep their old values field and UI users automatically are using the new field, without us having to build more complex logic in the FE.

cc @Waseem826

Reply:
If we are going for such a conversion, I'd prefer that the API does it. We can already retrieve the existing ApplicationInstallation there, set value to "" and valuesBlock to the new value.

Reply:
True that would also work without breaking existing GitOps Setups. Sounds good to me, I will add it to the dashboard PR and also add some explanation for the swagger there.